lib/engine/action/base.rb

```diff
@@ -75,6 +75,11 @@ def copy(game)
       def free?
         false
       end
+
+      def happened_before?(other)
```

Comment:
can you instead do 

def <=> 

to change the sorting, and then you can remove the usage of Time.at in the other section?

Reply:
Okay, done.

I already removed the old code from should_stop_applying_program in the original commit, are you referring to some other section?

Reply:
that's what i meant